components/SkilledImpact/elements/SkillContent.js

```diff
@@ -15,11 +21,28 @@ class SkillContent extends React.Component {
     this.state = {
       expanded: true,
       isAcceptingHelp: props.isAcceptingHelp,
-      skills: props.skills,
+      skills: props.currentUserProfile.skills,
+      loading: false,
       userId: props.userId,
     };
   }
 
+  componentDidMount() {
+    const { navigation } = this.props;
+    this.focusListener = navigation.addListener('didFocus', () => {
+      setTimeout(() => {
+        this.setState({
+          skills: this.props.currentUserProfile.skills,
+          loading: false,
+        });
+      }, 500);
```

Comment:
Not a fan of hacking around the issue like this. I think the actual issue is the lack of redux bindings in `components/SkilledImpact/SupporterSkilledImpactBody.js`. This state should be auto-updated by the `EDIT_PROFILE_SUCCESS` reducer, but we're not actually setting up Redux for that component.

I'm no Redux expert, but let me know if you can't figure it out and I can try to assist, or ask someone with more frontend experience than me.

Reply:
Hmm so I removed the timeout and somehow it still works now, which is odd since I remember it wasn't working before. If you also test it and it works, then great, otherwise I'm not really sure what else to do. Since `SkillContent.js` is connected to Redux and is the component that is returned to by the navigation, I was thinking the state would update fine, but honestly a lot of this is flying past my head and at this point after hours of staring at it I still don't really know how any of it _really_ works, so I'd greatly appreciate some help with it.

I tried multiple things like placing the Redux in the `SupporterSkilledImpactBody.js` or leaving it in the `SupporterSkilledImpactScreen.js` thinking that those would refresh after the state update but I don't think they ever did.

For now, I'll just push the changes that work for me right now to this PR (which included separating the `CampaignContent.js` into `SupporterCampaignContent.js` to support the different functionalities, since the tab was breaking when they were together)

Reply:
Also, as a side note, I'm getting a 500 response when editing the supporter profile with starting off with no skills (also getting it on master) but it works perfectly when I start with some skills and add/remove them.